tests/storage/test_devices.py

```diff
@@ -71,6 +71,28 @@ def test_get_devices_by_user(self):
             res["device2"],
         )
 
+    @defer.inlineCallbacks
+    def test_get_devices_by_remote(self):
```

Comment:
great start here. Could you also add a test for the case where somebody adds 101 devices?

suggest you do three calls to `add_device_change_to_streams`, with:
* add one device
* add 101 devices
* add one device

... we'll then expect to see the following results from `get_devices_by_remote`:
* 1 update
* empty list
* 1 update.